lib/ClassUtils.lua

```diff
@@ -51,13 +52,39 @@ function ClassUtils.makeConstructedClass(name, constructor)
 end
 
 function ClassUtils.makeEnum(keys)
-	return TableUtils.keyBy(
+	local enum =
+		TableUtils.keyBy(
 		keys,
 		function(key)
 			assert(key:match("^[A-Z_]+$"), "Enum keys must be defined as upper snake case")
 			return key
 		end
 	)
+
+	setmetatable(
+		enum,
+		{
+			__index = function(t, key)
+				error("Attempt to access key " .. key .. " which is not a valid key of the enum")
+			end,
+			__newindex = function(t, key)
+				error("Attempt to set key " .. key .. " on enum")
+			end
+		}
+	)
+
+	return enum
+end
+
+function ClassUtils.applyStrategyForEnum(enum, enumValue, strategies, ...)
```

Comment:
What do you think about
a) using a name that references switch, so it's use is more clear
b) currying the thing to allow delayed execution e.g. `getStrategySwitch(enum, strategies)(value, ...)